lib/cashaddr-browser.js

```diff
@@ -234,7 +249,8 @@ function deserialize(str, defaultPrefix) {
       data[i - prefixSize] = v;
   }
 
-  if (!chk.eqn(1))
+  const validChecksum = chk.eqn(1) && prefix === defaultPrefix;
```

Comment:
Should not we separate error messages so it's clear why it failed ?

Reply:
It's only possible when there is a prefix supplied in the address, otherwise it will fail the checksum. So the error is the same in both cases here.